tests/FSharp.Compiler.ComponentTests/EmittedIL/GenericComparison/Compare01.fsx.il.bsl

```diff
@@ -69,39 +69,45 @@
       IL_0003:  nop
       IL_0004:  ldc.i4.0
       IL_0005:  stloc.1
-      IL_0006:  br.s       IL_001b
+      IL_0006:  br.s       IL_0022
 
       IL_0008:  ldc.i4.1
       IL_0009:  ldc.i4.1
       IL_000a:  cgt
-      IL_000c:  stloc.2
```

Comment:
The increase in code size here has me a little perplaxed.  I'm assuming this is optimized code.  I'm not sure why the `cgt` hasn't been statically computed.

Reply:
The optimizer is matching on cgt when result is bool. I'm adding a new rule to return 1/0 when result is int32.
Same thing for clt. And we can add the same thing for any integer type.